### PR TITLE
Add test to LogStoreSuite for HDFSLogStore rename() error edge case

### PR DESCRIPTION
**Pull request by scottsand-db**
- follow up to https://github.com/delta-io/delta/pull/980
- rename `class HDFSLogStoreSuite` to `trait `HDFSLogStoreSuiteBase` so that internal (scala) and public (java) HDFSLogStore suites can both use that trait
- we add a specific test to `HDFSLogStoreSuiteBase` to test the `HDFSLogStore.writeInternal ... rename` edge case brought up by the above PR #980

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala [hunk 1]
```diff
@@ -220,9 +218,8 @@ class AzureLogStoreSuite extends LogStoreSuiteBase {
   protected def shouldUseRenameToWriteCheckpoint: Boolean = true
 }
 
-class HDFSLogStoreSuite extends LogStoreSuiteBase {
+trait HDFSLogStoreSuiteBase extends LogStoreSuiteBase {
```
Comment (zsxwing) [inline core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala:221]: QQ: why do we need a base class?

Comment (scottsand-db) [reply core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala:221]: it just makes more sense that we have some core functionality X. Then we want to test that with the internal (scala) hdfs impl and the external (java) hdsfs impl.

So `HDFSLogStoreSuite extends HDFSLogStoreSuiteBase` and `PublicHDFSLogStoreSuite extends HDFSLogStoreSuiteBase` is the cleanest way to do that.

If we had `PublicHDFSLogStoreSuite extends HDFSLogStoreSuite` (no `Base)` then that would look weird IMO.

**Review by zsxwing — APPROVED** [review]
LGTM
